org.civicrm.reqangular/test/services/file-upload_test.js

```diff
@@ -139,4 +139,55 @@ define([
       });
     });
   });
+
+  describe('FileUpload without mocked FileUploader service', function () {
```

Comment:
do you need any of this? you have a `FileUploaderMock` that you can extend by simply (i think, based on what i see you already did) adding
```js
return {
  // ..
  onCompleteItem: settings.onCompleteItem,
  onErrorItem: settings.onErrorItem,
  filters: settings.filters, // <-- added
};
```

this way you should be able to simply get the content of `uploader.filters` and run directly the `fileFormatFilter` function
```js
describe('allowed mime types', function () {
  var filterFn, allowedMimeTypes;

  beforeEach(function () {
    allowedMimeTypes = [ /* ... */ ];

    uploader = fileUpload.uploader({ allowedMimeTypes: allowedMimeTypes });
    filterFn = // find the `fileFormatFilter` function in `uploader.filters`
  });

  it('filters out files with not allowed mime types', function () {
    expect(filterFn(/* file allowed */)).toBe(true);
    expect(filterFn(/* file not allowed */)).toBe(false);
  });
});
```
so you don't need to use the real file service and you don't need to have two main branches in the test file